File: 02-week-2--recursion/05-day-5--recursive-selection-sort/ruby/selection_sort_recursive.rb
def selection_sort_recursive(arr)
  # find the lowest and put that in first position
  
  # minimum
  min = arr.min
  # index of minimum
  idx = arr.index(min)
  # deleting that index
  arr.delete_at(idx)
  # run this recursively
  result = selection_sort_recursive(arr)
  # return ^ to front of array
  result.unshift(min)
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: [-1, 2, 3, 5]"
  print "=> "
  print selection_sort_recursive([3, -1, 5, 2])

  puts

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: 02-week-2--recursion/05-day-5--recursive-selection-sort/javascript/selection_sort_recursive.js
function selectionSortRecursive(arr) {
  if (arr.length == 0){
    return []
  }
  // minimum
  const min = Math.min(arr)
  // idx
  const idx = arr.indexOf(min)
  // delete
  arr.splice(idx, 1)

  // initialize stack
  const result = selectionSortRecursive(arr)
  // populate array from bottom of stack up
  result.unshift(min)
  // return result
  return result
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: [-1, 2, 3, 5]");
  console.log("=>", selectionSortRecursive([3, -1, 5, 2]));

  console.log("");
}

module.exports = selectionSortRecursive;

// Please add your pseudocode to this file
// And a written explanation of your solution
